### ISSUE TITLE
Saving of rdf in TEI documents

### ISSUE THREAD

**Issue by ilovan** [cwrc/CWRC-Writer#479]
CWRC version of TEI has been customized to accept rdf before the TEI header.
This is not valid for canonical TEI, which doesn't seem to have a simple way of handling embedded rdf.

Possible options:

- xenoData: http://www.tei-c.org/release/doc/tei-p5-doc/en/html/ref-xenoData.html - TEI header child

OR:

- From the WissKI project:
It is clear from the TEI Guidelines that inserting RDF just between header and body is not at all conformant, but there have been discussions about inserting standoff annotations with an `<stdf>` extension to TEI.  They covered among other things feature structures, but not RDF, unfortunately.  (Oxygen also complains about the object-ids chosen, but this is a minor issue).  Another way to go, which I find less preferable, is the integration of RDFa as it has been done for HTML; this solution seems to me quite ponderous. I think a standoff-Tag which allows for RDF would be the optimal solution.  Two weeks ago I participated in a workshop by Greg Crane's group where we discussed the matter and there seem to be several people which share my opinion on `<stdf>`

Example documents: https://drive.google.com/file/d/0B6QNAc65ZX4PTDNmQkxBUHpPTHc/view?usp=sharing
https://drive.google.com/file/d/0B6QNAc65ZX4PZ2stYzN4LVlFYTA/view?usp=sharing

**Comment by ilovan** [comment]
Update: Susan consulted with James Cummings on this matter.
I am including his comments below for future reference:

There is basically two approaches that I've seen, one is to stash the RDF (often in RDF-XML) somewhere in the XML, the other is to try to convert it all to pure'ish TEI (even if in ad hoc constructs using semantically light elements).

Although there is a proposal for a purely standoff container It isn't even in the development branch of the TEI yet though Council has looked at it several times. I would ask Peter Stadler or Laurent Romary who would both be in a better place to discuss whether you should adopt it now or wait. (Let me know if you want me to put you in touch.) There are of course ways you could do everything that exists in that spec in pure TEI but it would be an ad hoc convention, hence why they are trying to nail it down.

<xenoData> was explicitly designed for people wanting to stash a copy of some non-TEI metadata somewhere and your RDF could indeed be stored there (even if it is really about annotations).  Frankly since you've already got quite a complex RDF setup I would probably just do this for convenience.

Another approach would be to look at the standoff proposal at https://github.com/laurentromary/stdfSpec but then use xenoData but ensuring it is stored in such a manner that it wouldn't be too hard to convert later to using something similar to that draft standoff spec.

**Comment by SusanBrown** [comment]
Yes and I am so talked it over with Syd Bauman who thought the xenodata approach was they way to go. So let's go with that.

Sent from my iPhone

On Jul 5, 2017, at 9:58 PM, Mihaela Ilovan <notifications@github.com<mailto:notifications@github.com>> wrote:


Update: Susan consulted with James Cummings on this matter.
I am including his comments below for future reference:

There is basically two approaches that I've seen, one is to stash the RDF (often in RDF-XML) somewhere in the XML, the other is to try to convert it all to pure'ish TEI (even if in ad hoc constructs using semantically light elements).

Although there is a proposal for a purely standoff container It isn't even in the development branch of the TEI yet though Council has looked at it several times. I would ask Peter Stadler or Laurent Romary who would both be in a better place to discuss whether you should adopt it now or wait. (Let me know if you want me to put you in touch.) There are of course ways you could do everything that exists in that spec in pure TEI but it would be an ad hoc convention, hence why they are trying to nail it down.

was explicitly designed for people wanting to stash a copy of some non-TEI metadata somewhere and your RDF could indeed be stored there (even if it is really about annotations). Frankly since you've already got quite a complex RDF setup I would probably just do this for convenience.

Another approach would be to look at the standoff proposal at https://github.com/laurentromary/stdfSpec but then use xenoData but ensuring it is stored in such a manner that it wouldn't be too hard to convert later to using something similar to that draft standoff spec.

**Comment by susanirenebrown** [comment]
Yes and I ALSO...
